### PR TITLE
Changelog 5.8: running changelog

### PR DESCRIPTION
**Pull request by oskosk**
* Updates changelog.txt to include 5.7's changelog
* Updates readme.txt with boilerplate for 5.8's changelog.
* Updates Stable tag to 5.7 (please update if there happens to exist a 5.7.1 release)

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -5,56 +5,47 @@ Stable tag: 5.7.1
 Requires at least: 4.7
 Tested up to: 4.9
 
-The one plugin you need for stats, related posts, search engine optimization, social sharing, protection, backups, speed, and email list management.
+Hassle-free design, marketing, and security — all in one place.
```
Comment (gibrown) [inline readme.txt]: We should not change this, it affects search results in the plugin directory. Also we should add "site search".

See https://wordpress.org/plugins/search/backups/ results for instance

File: readme.txt [hunk 2]
```diff
@@ -103,6 +94,46 @@ There are opportunities for developers at all levels to contribute. [Learn more
 
 == Changelog ==
 
+= 5.8 =
+
+* Release date: February 6, 2018
+* Release post: https://wp.me/p1moTy-731
+
+**Major Enhancements**
```
Comment (gibrown) [inline readme.txt]: Add search to Major Enhancements:

```
Search: Release from being in beta. Many improvements: customizer UX improvements; custom sorting support; post type filtering; and bug fixes.
```

File: readme.txt [hunk 3]
```diff
@@ -5,56 +5,47 @@ Stable tag: 5.7.1
...
-* [paid] Search engine optimization tools for Google, Bing, Twitter, Facebook, and WordPress.com
-* [paid] Advertising program that includes the best of AdSense, Facebook Ads, AOL, Amazon, Google AdX, and Yahoo
+= Marketing Services =
+Measure, promote, and earn moment from your site. Jetpack helps you with:
 
-= Security and Backup Services =
-Stop worrying about data loss, downtime, and hacking. Jetpack provides:
+* Site stats and analytics
+* Automated social media posting and scheduling in advance
+* Elasticsearch-powered related content and site search
```
Comment (gibrown) [inline readme.txt]: How about "Site search and filters that give great results and scales to millions of posts."

File: readme.txt (hunk 2 above)
Comment (jeherve) [reply readme.txt]: @gibrown I'm still working on the changelog items and testing list for Search ([quite a few more PRs to go through](https://github.com/Automattic/jetpack/pulls?page=1&q=is%3Apr+is%3Aclosed+label%3A%22%5BStatus%5D+Needs+Changelog%22&utf8=%E2%9C%93)). I'd be happy to get your feedback on it when it's ready! I've just committed what I have so far, and will continue to add to it later today. Feel free to add to the testing list as well, or ping me.

Thank you!

**Comment by jeherve** [comment]
@richardmuscat Could you review the feedback above about all the changes to the readme's wording? Thanks!

File: readme.txt (hunk 2 above)
Comment (gibrown) [reply readme.txt]: @jeherve np, sorry for jumping the gun. Came across it looking for something else. :)

**Pull request merged**